libraries/AP_Scripting/examples/Thrust_Stand.lua

```diff
@@ -0,0 +1,1339 @@
+
+  if not(_dev_initialised[TORQUE]) then
+      set_device(i2c_torque, TORQUE)
+      return init_nau7802, 100
+  end
+
+  -- Setup file to record data to
+  -- Scan through directory to find a unique file name.  Break up search over
+  -- with callbacks every 20 file names
+  if not(_file_index_found) then
+    for i = 1,20 do
+      if not (is_file(string.format(_file_name_fmt, _file_index+i))) then
+        -- Set file index to
+        _file_index = _file_index + i
+        _file_index_found = true
+        break
+      end
+
+      -- Reschedule init to continue searching for a file name
+      if i >= 20 then
+        _file_index = _file_index+20
+        return init, 100
+      end
+    end
+  end
+
+  -- local file = assert(io.open(file_name, "w"),"Could not make file: " .. file_name)
+  local file = assert(io.open(string.format(_file_name_fmt, _file_index), "w"), "Could not make file" .. string.format(_file_name_fmt, _file_index))
+  local header = 'Time (ms), Throttle (), RC Out (us), Motor Commutations (1/min), Voltage(V), Current (A), Thrust (g), Torque (kg.cm), ESC Volt (V), ESC Current (A), ESC RPM (rpm), ESC Temperature (deg C)\n'
+  file:write(header)
+  file:close()
+
+  -- Set first throttle step
+  set_next_thr_step()
+
+  _calibration_loaded_flag = load_calibration(THRUST)
+  if _calibration_loaded_flag then
+    gcs:send_text(4,"Thrust calibration values loaded ")
+    -- Advance system state, thrust calibration not needed
+    _sys_state = REQ_CAL_TORQUE_ZERO_OFFSET
+  end
+
+  _calibration_loaded_flag = _calibration_loaded_flag and load_calibration(TORQUE)
+  if _calibration_loaded_flag then
+    gcs:send_text(4,"Torque calibration values loaded ")
+    -- Advance system state, no calibration needed
+    _sys_state = DISARMED
+  end
+
+  -- Init neopixles
+  _led_chan = SRV_Channels:find_channel(98)
+  if not _led_chan then
+    gcs:send_text(6, "LEDs: channel not set")
+    return
+  end
+
+  -- Find_channel returns 0 to 15, convert to 1 to 16
+  _led_chan = _led_chan + 1
+
+  -- Added an extra LED to account for logic level shifter
+  if not serialLED:set_num_neopixel(_led_chan,  _num_leds+1) then
+    gcs:send_text(6, "LEDs: neopixle not set")
+  end
+
+  --- --- --- Set prerequisite parameters --- --- ---
+  -- Setup motor 1 so that we can apply servo overrides.  That way, if the lua script dies the 
+  -- override will time out and the motor will be stopped
+  local all_set = param:set_and_save('SERVO1_FUNCTION', 33)
+  all_set = all_set and param:set_and_save('MOT_SPIN_MIN', 0)
+  all_set = all_set and param:set_and_save('MOT_PWM_MIN', 1000)
+
+  -- Setup arming button function to arm and disarm copter aswell. This makes use of logging 
+  -- and standard motor protections whilst armed.
+  all_set = all_set and param:set_and_save('BTN_FUNC1', 41)
+
+  if all_set then
+    -- Now main loop can be started
+    return protected_update, 100
+  else
+    gcs:send_text(6, "Param set fail in init")
+    return
+  end
```

Comment:
```suggestion
  assert(all_set, "Param set fail in init")
```

Shorter and clearer then peter's suggestion. In practice though it's probably easier to just assert all your param set/saves above anyways where you do them, or just `assert(foo.set_and_save() and bar.set_and_save() and ... )`